akka-cluster-sharding-typed/src/main/scala/akka/cluster/sharding/typed/testkit/javadsl/EntityRef.scala

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright (C) 2019-2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.cluster.sharding.typed.testkit.javadsl
+
+import akka.actor.typed.ActorRef
+import akka.cluster.sharding.typed.internal.testkit.TestEntityRefImpl
+import akka.cluster.sharding.typed.javadsl.EntityRef
+import akka.cluster.sharding.typed.javadsl.EntityTypeKey
+import akka.util.unused
+
+/**
+ * For testing purposes this `EntityRef` can be used in place of a real [[EntityRef]].
+ * It forwards all messages to the `probe`.
+ */
+object TestEntityRef {
+  def of[M](@unused typeKey: EntityTypeKey[M], entityId: String, probe: ActorRef[M]): EntityRef[M] =
```

Comment:
this looks find but we sometimes get bitten for not having at least compile only tests for javadsl